src/spanningtrees/kruskal.jl

```diff
@@ -0,0 +1,55 @@
+immutable KruskalHeapEntry{T<:Real}
+    edge::LightGraphs.Edge
+    dist::T
+end
+
+isless(e1::KruskalHeapEntry, e2::KruskalHeapEntry) = e1.dist < e2.dist
+
+""" Performs [Quick-Find algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure)
+on a given pair of nodes `p`and `q`, and makes a connection between them
+in the vector `nodes`.
+"""
+function quick_find!(nodes, p, q)
+    pid = nodes[p]
+    qid = nodes[q]
+    for i in 1:length(nodes)
+        if nodes[i] == pid
+            nodes[i] = qid
+        end
+    end
+end
+
+"""Performs [Kruskal's algorithm](https://en.wikipedia.org/wiki/Kruskal%27s_algorithm)
+on a connected, non-directional graph `g`, having adjacency matrix `distmx`,
+and computes minimum spanning tree. Returns a `Vector{KruskalHeapEntry}`,
+that contains the containing edges and its weights.
+"""
+function kruskal_mst{T<:Real}(
+    g::SimpleGraph,
+    distmx::AbstractArray{T, 2}
```

Comment:
`distmx::AbstractArray{T, 2} = DefaultDistance()`

so that we can call kruskal_mst(g)  and assume weight 1 on each edge. Take a look to src/distance.jl for examples